File: Advanced Algorithms Sem 6/amortized analysis/aggregate_multiPop.cpp
#include <iostream>
#include <bits/stdc++.h>
#include <vector>

using namespace std;

void print(vector<int> s)
{
    for (int i = 0; i < s.size(); i++)
    {
        cout << s[i] << " ";
    }
    cout<<endl;
}

int main()
{
    vector<int> s;
   
    int arr[]={8,6,7,5,2,6,7,1};
    int total=0;
   
    for(int i=0;i<8;i++)
    {
        if(arr[i]<=s.size())
        {
            for(int j=0;j<arr[i];j++)
            {
                s.pop_back();
            }
            s.push_back(arr[i]);
           
            print(s);
            cout<<"Cost = "<<arr[i]+1<<endl;
            total=total+arr[i]+1;
        }
        else
        {
            s.push_back(arr[i]);
            print(s);
            cout<<"Cost = 1"<<endl;
            total=total+1;
        }
    }
   
    float a_cost=float(total)/8;
   
    cout<<"Total cost = "<<total<<endl;
    cout<<"Ammortized cost = "<<a_cost;

    return 0;
}

File: Advanced Algorithms Sem 6 for pracs/amortized analysis/aggregate_multipop.py
stack = []
list = [8,6,7,5,2,6,7,1]
total = 0

for i in range(len(list)):
    if(list[i]<=len(stack)):
        for j in range(list[i]):
            stack.pop()
        stack.append(list[i])

        print(stack)
        print("Cost = ",list[i]+1)
        total+=list[i]+1

    else:
        stack.append(list[i])
        print(stack)
        print("Cost = 1")
        total+=1


print("Total Cost = ", total)
print("Ammortized Cost = ", total/len(list))
